tests/unit/resource/request/request_service_dereference_resource_metadata_test.go

```diff
@@ -0,0 +1,258 @@
+						DidString:        testconstants.NotExistentTestnetDid,
+						MethodSpecificId: testconstants.NotExistentIdentifier,
+						Method:           testconstants.ValidMethod,
+					},
+				},
+				ContentStream: nil,
+				Metadata:      &types.ResolutionDidDocMetadata{},
+			},
+			expectedError: types.NewNotFoundError(testconstants.NotExistentTestnetDid, types.DIDJSONLD, nil, false),
+		},
+	),
+
+	Entry(
+		"cannot get resource metadata with an invalid DID",
+		resourceMetadataTestCase{
+			didURL: fmt.Sprintf(
+				"/1.0/identifiers/%s/resources/%s/metadata",
+				testconstants.InvalidDid,
+				testconstants.ExistentResourceId,
+			),
+			resolutionType: types.DIDJSONLD,
+			expectedDereferencingResult: &DereferencingResult{
+				DereferencingMetadata: &types.DereferencingMetadata{
+					DidProperties: types.DidProperties{
+						DidString:        testconstants.InvalidDid,
+						MethodSpecificId: testconstants.InvalidIdentifier,
+						Method:           testconstants.InvalidMethod,
+					},
+				},
+				ContentStream: nil,
+				Metadata:      &types.ResolutionDidDocMetadata{},
+			},
+			expectedError: types.NewMethodNotSupportedError(testconstants.InvalidDid, types.DIDJSONLD, nil, false),
+		},
+	),
+
+	Entry(
+		"cannot get resource metadata with an existent DID, but not existent resourceId",
+		resourceMetadataTestCase{
+			didURL: fmt.Sprintf(
+				"/1.0/identifiers/%s/resources/%s/metadata",
+				testconstants.ExistentDid,
+				testconstants.NotExistentIdentifier,
+			),
+			resolutionType: types.DIDJSONLD,
+			expectedDereferencingResult: &DereferencingResult{
+				DereferencingMetadata: &types.DereferencingMetadata{
+					DidProperties: types.DidProperties{
+						DidString:        testconstants.ExistentDid,
+						MethodSpecificId: testconstants.ValidIdentifier,
+						Method:           testconstants.ValidMethod,
+					},
+				},
+				ContentStream: nil,
+				Metadata:      &types.ResolutionDidDocMetadata{},
+			},
+			expectedError: types.NewNotFoundError(testconstants.ExistentDid, types.DIDJSONLD, nil, true),
+		},
+	),
+
+	Entry(
+		"cannot get resource metadata with an existent DID, but an invalid resourceId",
+		resourceMetadataTestCase{
+			didURL: fmt.Sprintf(
+				"/1.0/identifiers/%s/resources/%s/metadata",
+				testconstants.ExistentDid,
+				testconstants.InvalidIdentifier,
+			),
+			resolutionType: types.DIDJSONLD,
+			expectedDereferencingResult: &DereferencingResult{
+				DereferencingMetadata: &types.DereferencingMetadata{
+					DidProperties: types.DidProperties{
+						DidString:        testconstants.ExistentDid,
+						MethodSpecificId: testconstants.ValidIdentifier,
+						Method:           testconstants.ValidMethod,
+					},
+				},
+				ContentStream: nil,
+				Metadata:      &types.ResolutionDidDocMetadata{},
+			},
+			expectedError: types.NewInvalidDidUrlError(testconstants.ExistentDid, types.DIDJSONLD, nil, true),
+		},
+	),
+
+	Entry(
+		"cannot get resource metadata with an invalid representation",
```

Comment:
Let's rename `representation` to `contentType`

Reply:
Done! ✅